presto-main/src/main/java/com/facebook/presto/sql/gen/BytecodeUtils.java

```diff
@@ -181,45 +180,85 @@ public static BytecodeNode generateInvocation(Scope scope, String name, ScalarFu
         // Index of parameter (without @IsNull) in Presto function
         int realParameterIndex = 0;
 
+        // Go through all the choices in the function and then pick the best one
+        List<ScalarImplementationChoice> choices = function.getAllChoices();
+        ScalarImplementationChoice bestChoice = choices.get(0);
```

Comment:
I would suggest not to assign this. And add a `checkState` to assert `bestChoice` is not null after the for loop. 

This allows us to do sanity check over the scalar implementation: i.e. at least one valid implementation should be provided.